### PR TITLE
installer/server/terraform: add an option to retry `terraform apply`

### PR DESCRIPTION
**Pull request by Quentin-M**
This PR introduces a new `retry` boolean to the `terraform/apply` API endpoint. When specified, the backend will try to restore the executor from the session rather than creating a new one. If there was no executor in the session, an error will be thrown.

### CODE REVIEW

File: installer/server/terraform.go [hunk 1]
```diff
@@ -162,10 +163,20 @@ func terraformApplyHandler(sessionProvider sessions.Store) ctxh.ContextHandler {
 		}
 		defer req.Body.Close()
 
-		// Create a new TerraForm Executor with the TF variables.
-		ex, errCtx := newExecutorFromApplyHandlerInput(&input)
-		if errCtx != nil {
-			return errCtx
+		var ex *terraform.Executor
+		var errCtx *ctxh.AppError
+		if !input.Retry {
```
Comment (ggreer) [inline installer/server/terraform.go]: Nitpick: If you have an else, why invert the conditional? Just swap the bodies.

File: installer/server/terraform.go [hunk 2]
```diff
@@ -162,10 +163,20 @@ func terraformApplyHandler(sessionProvider sessions.Store) ctxh.ContextHandler {
...
+			// Create a new TerraForm Executor with the TF variables.
+			ex, errCtx = newExecutorFromApplyHandlerInput(&input)
+			if errCtx != nil {
+				return errCtx
+			}
+		} else {
+			// Restore the execution environment from the session.
+			_, ex, _, errCtx = restoreExecutionFromSession(req, sessionProvider, &input.Credentials)
+			if errCtx != nil {
```
Comment (ggreer) [inline installer/server/terraform.go]: I think errCtx check can be moved out of the if/else, reducing repetition. Code golf! ⛳️

Comment (Quentin-M) [inline installer/server/terraform.go]: Yeap but typically had the exact opposite review on projects like Clair for readability reasons.

File: installer/server/terraform.go (hunk 1 above)
Comment (Quentin-M) [inline installer/server/terraform.go]: Just to have the usual case first. Can swap if you want.

**Pull request merged**